### PR TITLE
HWKALERTS-233 : Add trigger enable/disable to REST API

### PR DESCRIPTION
**Pull request by jshaughn**
- new endpoints triggers/enabled and triggers/groups/enabled
- update existing itests to use new endpoints where applicable
- add new test

### CODE REVIEW

**Comment by jshaughn** [comment]
@lucasponce Please review as time permits.

File: hawkular-alerts-rest-tests/src/test/groovy/org/hawkular/alerts/rest/ActionsITest.groovy [hunk 1]
```diff
@@ -333,9 +333,7 @@ class ActionsITest extends AbstractITestBase {
         assertEquals(1, resp.data.size())
 
         // ENABLE Trigger
-        testTrigger.setEnabled(true);
-
-        resp = client.put(path: "triggers/test-email-threshold", body: testTrigger)
+        resp = client.put(path: "triggers/enabled", query:[triggerIds:"test-email-threshold",enabled:true] )
```
Comment (lucasponce) [inline hawkular-alerts-rest-tests/src/test/groovy/org/hawkular/alerts/rest/ActionsITest.groovy]: We should maintain tests with the old way to disable (using the whole trigger update) to not miss possible regressions.
I see there were several modifications in tests, I couldn't see if all of them are updated with this one.
Let's maintain explicit tests for non-groups and groups triggers.

Comment (jshaughn) [inline hawkular-alerts-rest-tests/src/test/groovy/org/hawkular/alerts/rest/ActionsITest.groovy]: I'm not sure I understand the regression risk?  The point of the new endpoints is to reduce risk (and add convenience) when just performing an enable or disable.  I have updated all of these in the existing tests.  I can go back and undo those changes but can you better explain why that is necessary?

Comment (lucasponce) [inline hawkular-alerts-rest-tests/src/test/groovy/org/hawkular/alerts/rest/ActionsITest.groovy]: Yes. If we eliminate all tests when we enable/disable a trigger using the normal update endpoint, we might miss a explicit test for testing the previous (and currently used) behaviour.
So, those changes are not bad, but we should maintain some tests to enable/disable triggers (non-group and groups) using the current and previous whole trigger update.

Comment (jshaughn) [inline hawkular-alerts-rest-tests/src/test/groovy/org/hawkular/alerts/rest/ActionsITest.groovy]: I'll change a few back to have a mix although I had updated only the instances where the only change to the already stored trigger was the enabled flag, just a put of that one change.

**Comment by lucasponce** [comment]
Thanks @jshaughn

**Pull request merged**